lib/db/schema/patch-022-023.sql

```diff
@@ -0,0 +1,140 @@
+-- This migration:
+--
+-- * Changes the type of the devices.callbackPublicKey column
+-- * Add a new devices.callbackAuthKey column
+-- * Updates the accountDevices stored procedure with the new column
+-- * Alters three stored procedures, createDevice, updateDevice and
+--   sessionWithDevice to work with the new columns
+
+-- 2-steps callbackPublicKey change: we avoid junk values, and we don't block
+-- concurent queries.
+
+ALTER TABLE devices
+CHANGE callbackPublicKey oldCallbackPublicKey BINARY(32),
+ADD COLUMN callbackPublicKey VARCHAR(88),
+ADD COLUMN callbackAuthKey VARCHAR(24);
+
+ALTER TABLE devices
+DROP COLUMN oldCallbackPublicKey;
```

Comment:
I tried these out on a dev box, and they seem to apply cleanly, and the old stored procedures appear to keep working even with the change in datatype.  So I'm feeling pretty good about this migration 😄 

Could you please add ` ALGORITHM = INPLACE, LOCK = NONE` to the end of both `ALTER` statements?  For example, the second one should become:

```
ALTER TABLE devices DROP COLUMN oldCallbackPublicKey, ALGORITHM = INPLACE, LOCK = NONE
```

I tried them with that addition and they worked fine, but it's worth having on there explicitly in case the production MySQL decides that it e.g. needs to take an exclusive lock for whatever reason.  With those declarations on, the any alters likely to produce a production outage will fail like this:

```
mysql> ALTER TABLE devices MODIFY callbackPublicKey BINARY(31), ALGORITHM = INPLACE, LOCK = NONE;
ERROR 1846 (0A000): ALGORITHM=INPLACE is not supported. Reason: Cannot change column type INPLACE. Try ALGORITHM=COPY.
```